iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergTableOptimizer.java

```diff
@@ -79,29 +91,36 @@ public Set<CompactionInfo> findPotentialCompactions(long lastChecked, ShowCompac
     getTables().stream()
         .filter(table -> !skipDBs.contains(table.getDb()))
         .filter(table -> !skipTables.contains(table.getNotEmptyDbTable()))
-        .map(table -> Pair.of(table, resolveMetastoreTable(table.getNotEmptyDbTable())))
-        .filter(tablePair -> MetaStoreUtils.isIcebergTable(tablePair.getValue().getParameters()))
-        .filter(tablePair -> {
-          long currentSnapshotId = Long.parseLong(tablePair.getValue().getParameters().get("current-snapshot-id"));
-          Long cachedSnapshotId = snapshotIdCache.get(tablePair.getKey().getNotEmptyDbTable());
-          return cachedSnapshotId == null || cachedSnapshotId != currentSnapshotId;
+        .map(table -> {
+          org.apache.hadoop.hive.ql.metadata.Table hiveTable = getHiveTable(table.getDb(), table.getTable());
+          org.apache.iceberg.Table icebergTable = IcebergTableUtil.getTable(conf, hiveTable.getTTable());
+          return Triple.of(table, hiveTable, icebergTable);
```

Comment:
You don't need Triple. `Pair.of(hiveTable, icebergTable)` should be enough
t.getLeft().getFullTableName() returns the TableName

Reply:
Done